core/src/main/java/org/opensearch/sql/datasource/model/DataSourceMetadata.java

```diff
@@ -44,18 +57,32 @@ public class DataSourceMetadata {
 
   @JsonProperty private String resultIndex;
 
+  public static Function<String, String> DATASOURCE_TO_RESULT_INDEX =
+      datasourceName -> String.format("%s_%s", DEFAULT_RESULT_INDEX, datasourceName);
+
   public DataSourceMetadata(
       String name,
+      String description,
       DataSourceType connector,
       List<String> allowedRoles,
       Map<String, String> properties,
       String resultIndex) {
+    String errorMessage = validateCustomResultIndex(resultIndex);
+
     this.name = name;
     this.connector = connector;
-    this.description = StringUtils.EMPTY;
+    this.description = description;
     this.properties = properties;
     this.allowedRoles = allowedRoles;
-    this.resultIndex = resultIndex;
+
+    if (errorMessage != null) {
```

Comment:
Minor Nit: can we move this up, In case there is a new revision.

Reply:
yes, will do